### DISCUSSION TITLE
How To: Tool which uses GPT4 chat completion

### DISCUSSION THREAD

**Discussion by WebsheetPlugin** [microsoft/autogen#2363] [Q&A]
I am not sure how to do it in Autogen, but I am looking for an easy way, to call via OpenAI's GPT4 chat completion with some specific instructions, then still inside the tool, I want to use this information to do some processes and only then return some result from the tool.

I have checked "oai.client", but this looks like the old completions and not the new chat completion API.

Or can I use it like https://platform.openai.com/docs/guides/text-generation/chat-completions-api here?

```
from openai import OpenAI
client = OpenAI()

response = client.chat.completions.create(
  model="gpt-3.5-turbo",
  messages=[
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": "Who won the world series in 2020?"},
    {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."},
    {"role": "user", "content": "Where was it played?"}
  ]
)
```

And it will still allow me to still use:

1. get_total_usage
2. config_list 
3. autogen.runtime_logging

Also, these nested chats seem way too complicated for me. For me, it's not really clear why I should use it in this case. As I am not looking for some kind of inner conversation.

**Comment by sonichi** [comment]
You can do:
```python
from autogen import OpenAIWrapper

client = OpenAIWrapper()
response = client.create(
  model="gpt-3.5-turbo",
  messages=[
    {"role": "system", "content": "You are a helpful assistant."},
    {"role": "user", "content": "Who won the world series in 2020?"},
    {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."},
    {"role": "user", "content": "Where was it played?"}
  ]
)
```
For the nested chat discussion, please refer to #2223

**Reply by sonichi** [reply]
`config_list` can be used in `OpenAIWrapper(config_list=config_list)`
